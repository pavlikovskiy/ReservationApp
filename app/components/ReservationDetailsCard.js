/**
 * @flow
 */

import React, { Component } from "react";
import { Text, View, StyleSheet } from "react-native";

export default class ReservationDetailsCard extends Component<ReservationDetailsCardProps> {
  render() {
    const { reservation } = this.props;

    return (
      <View key={reservation.id} style={styles.searchResultContainer}>
        <Text style={styles.header}>
          {reservation.name} at {reservation.hotelName}
        </Text>
        <Text style={styles.label}>Name: {reservation.name}</Text>
        <Text style={styles.label}>Hotel Name: {reservation.hotelName}</Text>
        <Text style={styles.label}>From: {reservation.arrivalDate}</Text>
        <Text style={styles.label}>To: {reservation.departureDate}</Text>
      </View>
    );
  }
}

type ReservationDetailsCardProps = {
  reservation: {
    id: string,
    name: string,
    hotelName: string,
    arrivalDate: string,
    departureDate: string
  }
};

const styles = StyleSheet.create({
  searchResultContainer: {
    flex: 1,
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
    backgroundColor: "#FAFAFA",
    paddingLeft: 20,
    paddingBottom: 20
  },
  header: {
    flex: 1,
    paddingVertical: 10,
    paddingHorizontal: 10,
    paddingBottom: 20,
    paddingTop: 10,
    color: "#808080",
    fontSize: 14,
    alignItems: "center",
    justifyContent: "center",
    fontWeight: "bold"
  },
  label: {}
});
