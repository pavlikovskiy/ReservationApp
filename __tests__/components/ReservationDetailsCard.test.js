import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import ReservationDetailsCard from "../../app/components/ReservationDetailsCard";

describe("ReservationDetailsCard", () => {
  it("should render without issues", () => {
    const reservation = {
      id: "id",
      name: "name",
      hotelName: "hotelName",
      arrivalDate: "today",
      departureDate: "tomorrow"
    };

    const component = shallow(
      <ReservationDetailsCard reservation={reservation} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
