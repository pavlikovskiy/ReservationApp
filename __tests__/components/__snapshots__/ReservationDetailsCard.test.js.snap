// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationDetailsCard should render without issues 1`] = `
<View
  key="id"
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 20,
      "paddingLeft": 20,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#808080",
        "flex": 1,
        "fontSize": 14,
        "fontWeight": "bold",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingHorizontal": 10,
        "paddingTop": 10,
        "paddingVertical": 10,
      }
    }
  >
    name
     at 
    hotelName
  </Text>
  <Text
    style={Object {}}
  >
    Name: 
    name
  </Text>
  <Text
    style={Object {}}
  >
    Hotel Name: 
    hotelName
  </Text>
  <Text
    style={Object {}}
  >
    From: 
    today
  </Text>
  <Text
    style={Object {}}
  >
    To: 
    tomorrow
  </Text>
</View>
`;
